detection-rules/impersonation_recipient_domain.yml

```diff
@@ -0,0 +1,35 @@
+name: "Impersonation using recipient domain (first-time sender)"
+description: |
+  The recipient's domain is used in the sender's display name
+  in order to impersonate the organization. The impersonation has been 
+  observed to use both the recipient's full email address, as well as 
+  just the domain.
+type: "rule"
+severity: "medium"
```

Comment:
I've seen this deliver anything from HTML smuggling to credential phishing to spam. Since this rule isn't designed to detect any one of those specifically, "high" doesn't feel appropriate